policies/recipes/steps.go

```diff
@@ -131,27 +132,100 @@ func StepFileExec(step *osconfigpb.SoftwareRecipe_Step_FileExec, artifacts map[s
 
 // StepScriptRun builds the command for a ScriptRun step
 func StepScriptRun(step *osconfigpb.SoftwareRecipe_Step_ScriptRun, artifacts map[string]string, runEnvs []string, stepDir string) error {
+	if runtime.GOOS == "windows" {
+		switch step.ScriptRun.Interpreter {
+		case osconfigpb.SoftwareRecipe_Step_RunScript_INTERPRETER_UNSPECIFIED:
+			return scriptRunDirect(step, runEnvs, stepDir)
+		case osconfigpb.SoftwareRecipe_Step_RunScript_SHELL:
+			return scriptRunCmd(step, runEnvs, stepDir)
+		case osconfigpb.SoftwareRecipe_Step_RunScript_POWERSHELL:
+			return scriptRunPowershell(step, runEnvs, stepDir)
+		default:
+			return fmt.Errorf("invalid interpreter %q", step.ScriptRun.Interpreter)
+		}
+	}
+
+	switch step.ScriptRun.Interpreter {
+	case osconfigpb.SoftwareRecipe_Step_RunScript_INTERPRETER_UNSPECIFIED:
+		return scriptRunDirect(step, runEnvs, stepDir)
+	case osconfigpb.SoftwareRecipe_Step_RunScript_SHELL:
+		return scriptRunSh(step, runEnvs, stepDir)
+	case osconfigpb.SoftwareRecipe_Step_RunScript_POWERSHELL:
+		return fmt.Errorf("interpreter %q cannot be used on non-Windows system", step.ScriptRun.Interpreter)
+	default:
+		return fmt.Errorf("invalid interpreter %q", step.ScriptRun.Interpreter)
+	}
+}
+
+func scriptRunSh(step *osconfigpb.SoftwareRecipe_Step_ScriptRun, runEnvs []string, stepDir string) error {
 	scriptPath := filepath.Join(stepDir, "script")
 
-	f, err := os.Create(scriptPath)
+	err := writeScript(scriptPath, step.ScriptRun.Script)
 	if err != nil {
 		return err
 	}
 
-	f.WriteString(step.ScriptRun.Script)
-	f.Close()
-	if err := os.Chmod("/tmp/scriptrun", 0755); err != nil {
+	var qargs []string
+	for _, arg := range step.ScriptRun.Args {
+		qargs = append(qargs, fmt.Sprintf("%q", arg))
+	}
+	command := scriptPath + " " + strings.Join(qargs, " ")
+	args := []string{"-c", command}
+	return executeCommand("/bin/sh", stepDir, runEnvs, args...)
+}
+
+func scriptRunDirect(step *osconfigpb.SoftwareRecipe_Step_ScriptRun, runEnvs []string, stepDir string) error {
+	scriptPath := filepath.Join(stepDir, "script")
+
+	err := writeScript(scriptPath, step.ScriptRun.Script)
+	if err != nil {
+		return err
+	}
+
+	return executeCommand(scriptPath, stepDir, runEnvs, step.ScriptRun.Args...)
+}
+
+func scriptRunPowershell(step *osconfigpb.SoftwareRecipe_Step_ScriptRun, runEnvs []string, stepDir string) error {
+	scriptPath := filepath.Join(stepDir, "script.ps1")
+
+	err := writeScript(scriptPath, step.ScriptRun.Script)
+	if err != nil {
+		return err
+	}
+
+	args := append([]string{"-File", scriptPath}, step.ScriptRun.Args...)
+	return executeCommand("PowerShell", stepDir, runEnvs, args...)
+}
+
+func scriptRunCmd(step *osconfigpb.SoftwareRecipe_Step_ScriptRun, runEnvs []string, stepDir string) error {
+	scriptPath := filepath.Join(stepDir, "script.bat")
```

Comment:
as far as I can tell on windows it needs the right file extension to be able to run correctly on windows, is that correct?